### PR TITLE
SMP consent endpoint for investors to verify SMP access before real call

### PR DESCRIPTION
**Pull request by chaluja7**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -754,6 +754,31 @@ Purchase a share of an investment offer on the secondary marketplace
...
++ Response 200
+
+    + Body
+
+        ```json
+        {
+            "smpConsent": {
+                "agreedOn": "2020-01-28T21:47:33.339+01:00"
+            }
+        }
```
Comment (triceo) [inline apiary.apib:779]: How does it look when the user hasn't given the consent? Is the response empty?

Comment (triceo) [reply apiary.apib:779]: Also, there is an `/investors/me/restrictions` endpoint, which has a `cannotAccessSmp` boolean field. What's that for, and what is the difference?

Comment (chaluja7) [reply apiary.apib:779]: > How does it look when the user hasn't given the consent? Is the response empty?

Yes, empty body response with HTTP status 200

**Review by lukasnovy — APPROVED** [review]

File: apiary.apib (hunk 1 above)
Comment (chaluja7) [reply apiary.apib:779]: > Also, there is an `/investors/me/restrictions` endpoint, which has a `cannotAccessSmp` boolean field. What's that for, and what is the difference?

`cannotAccessSmp` is internally used to allow / deny access to SMP and has nothing to do with the fact whether the user agreed on the consent or not.

**Pull request merged**